src/hotspot/cpu/x86/vm_version_x86.hpp

```diff
@@ -911,6 +911,14 @@ enum Extended_Family {
   static bool is_intel_skylake() { return is_intel_family_core() &&
                                           extended_cpu_model() == CPU_MODEL_SKYLAKE; }
 
+  // avx3_threshold() sets the threshold at which 64-byte instructions are used
+  // for implementing the array copy and clear operations.
+  // The Intel platforms that supports the serialize instruction
+  // has improved implementation of 64-byte load/stores and so the threshold
+  // is set to 0 for these platforms.
+  static int avx3_threshold() { return ((is_intel_family_core() &&
+                                supports_serialize()) ? 0: AVX3Threshold); }
```

Comment:
Hi @sviswa7 , Should we not return a zero threshold only if user does not explicitly set AVX3Threshold i.e. in default case.

Reply:
@jatin-bhateja On these platforms it is beneficial to set the threshold to zero for copy and clear operations and hence the override. I have described that in the comment in detail as well.

Reply:
@sviswa7 Even if these ISAs have a more efficient copy/clear operations, it still seems incorrect to override a user’s value for AVX3Threshold.  They may have other reasons to set it (eg maybe they’re just benchmarking to see the difference in speed).  So 0 should be returned only if `FLAG_IS_DEFAULT(AVX3Threshold)` is also true, methinks?

Reply:
@sviswa7 I tend to agree with @jatin-bhateja . AVX3Threshold is a diagnostic flag so if someone has deliberately modified it so they can measure something, your change will make that impossible on newer systems. You may want to define a static field to store the actual value for `avx3_threshold()` to return and initialize it during VM initialization. Or lazy initialize it on first use.